lib/nfp

```diff
@@ -145,66 +146,156 @@ function create_nfp_gbp_resources {
 
 }
 
+function configure_visibility_user_data { 
+CUR_DIR=$PWD
+TOP_DIR=$4
+sudo cp -r $TOP_DIR/devstack/exercises/nfp/visibility_user_data/visibility-user-data /opt/.
+cd /opt
+sudo rm -rf my.key my.key.pub
+visibility_vm_ip=$2
+HOST_IP=$1
+MYSQL_PASSWORD=$3
+sudo ssh-keygen -t rsa -N "" -f my.key
+value=`sudo cat my.key.pub`
+sudo echo  $value
+sudo sed -i "8 i\      -\ $value" visibility-user-data
+sudo sed -i '9d' visibility-user-data
+sudo sed -i "s/visibility_vm_ip=*.*/visibility_vm_ip=$visibility_vm_ip/g" visibility-user-data
+sudo sed -i "s/sc_active_ip=*.*/sc_active_ip=$HOST_IP/g" visibility-user-data
+sudo sed -i "s/sc_standby_ip=*.*/sc_standby_ip=$HOST_IP/g" visibility-user-data
+sudo sed -i "s/service_controller_vip=*.*/service_controller_vip=$HOST_IP:8080/g" visibility-user-data
+sudo sed -i "s/db_user=*.*/db_user=root/g" visibility-user-data
+sudo sed -i "s/db_password=*.*/db_password=$MYSQL_PASSWORD/g" visibility-user-data
+sudo sed -i "s/db_server_ip=*.*/db_server_ip=$HOST_IP/g" visibility-user-data
+sudo sed -i "s/statsd_host=*.*/db_server_ip=$visibility_vm_ip/g" visibility-user-data
+sudo sed -i "s/statsd_port=*.*/statsd_port=8125/g" visibility-user-data
+sudo sed -i "s/rsyslog_host=*.*/rsyslog_host=$visibility_vm_ip/g" visibility-user-data
+sudo sed -i "s/rabbit_host=*.*/rabbit_host=$visibility_vm_ip/g" visibility-user-data
+sudo sed -i "s/rabbit_port=*.*/rabbit_port=5672/g" visibility-user-data
+sudo sed -i "s/rsyslog_host=*.*/rsyslog_host=$visibility_vm_ip/g" visibility-user-data
+cd $CUR_DIR
+
+}
 function upload_images_and_launch_configuratorVM {
     
     TOP_DIR=$1
     source $TOP_DIR/openrc neutron service
-    ConfiguratorQcow2ImageName=$2
-    VyosQcow2Image=$3
-    HaproxyQcow2Image=$4
-    ConfiguratorImageName=configurator
-    VyosImageName=vyos
-    HaproxyImageName=haproxy
-    if [ ! -z "$1" -a "$1" != " " ]; then
-        echo "Uploading Image : $ConfiguratorImageName"
-        glance image-create --name $ConfiguratorImageName --disk-format qcow2  --container-format bare  --visibility public --file $ConfiguratorQcow2ImageName
-        echo "Uploading Image : $VyosImageName"
-        glance image-create --name $VyosImageName --disk-format qcow2  --container-format bare  --visibility public --file $VyosQcow2Image
-        echo "Uploading Image : $HaproxyImageName"
-        glance image-create --name $HaproxyImageName --disk-format qcow2  --container-format bare  --visibility public --file $HaproxyQcow2Image
+    [[ $5 = "" ]] && VISIBILITY_FLAG=False
+    if [[ $VISIBILITY_FLAG = False ]]; then
```

Comment:
We need to support launching two different VMs for configurator and visibility later.